src/blackdoor/cqbe/cli/dh256.java

```diff
@@ -172,24 +234,29 @@ public static void insert(String[] args) {
 							.println("specified file does not exist or we lack read permissions");
 					return;
 				}
-				Router router;
-				CASFileAddress fileAddress = new CASFileAddress(file);
-				byte[] fileBytes = Files.readAllBytes(file.toPath());
-				if (out.containsKey("bootstrap")) {
-					String bootstrap = out.get("bootstrap").getValues().get(0);
-					String[] bootstrapSplit = bootstrap.split(":");
-					L3Address a = new L3Address(
-							InetAddress.getByName(bootstrapSplit[0]),
-							Integer.parseInt(bootstrapSplit[1]));
-					router = Router.fromBootstrapNode(a);
-				} else {
-					router = Router.fromDefaultLocalNode();
+				if (!out.containsKey("check")) {
+					Router router;
+					CASFileAddress fileAddress = new CASFileAddress(file);
+					byte[] fileBytes = Files.readAllBytes(file.toPath());
+					if (out.containsKey("bootstrap")) {
+						String bootstrap = out.get("bootstrap").getValues()
+								.get(0);
+						String[] bootstrapSplit = bootstrap.split(":");
+						L3Address a = new L3Address(
+								InetAddress.getByName(bootstrapSplit[0]),
+								Integer.parseInt(bootstrapSplit[1]));
+						router = Router.fromBootstrapNode(a);
+					} else {
+						router = Router.fromDefaultLocalNode();
+					}
+					int numberStored = router.put(fileAddress, fileBytes);
+					if (numberStored > 0)
+						System.out.println("file added successfully by "
+								+ numberStored + " nodes with address: "
```

Comment:
awk/confusing phrasing
"file with address <> successfasdfasly stored by x nodes" might be better